types/benches/bytesrepr_bench.rs

```diff
@@ -437,6 +446,300 @@ fn deserialize_u512(b: &mut Bencher) {
+    Account::new(
+        account_hash,
+        named_keys,
+        main_purse,
+        associated_keys,
+        action_thresholds,
+    )
+}
+
+fn serialize_account(b: &mut Bencher) {
+    let account = generate_account(10, 10);
+    b.iter(|| ToBytes::to_bytes(black_box(&account)));
+}
+
+fn deserialize_account(b: &mut Bencher) {
+    let account = generate_account(10, 10);
+    let account_bytes = Account::to_bytes(&account).unwrap();
+    b.iter(|| Account::from_bytes(black_box(&account_bytes)).unwrap());
+}
+
+fn serialize_contract(b: &mut Bencher) {
+    let contract = generate_contract(10, 10);
+    b.iter(|| ToBytes::to_bytes(black_box(&contract)));
+}
+
+fn deserialize_contract(b: &mut Bencher) {
+    let contract = generate_contract(10, 10);
+    let contract_bytes = Contract::to_bytes(&contract).unwrap();
+    b.iter(|| Contract::from_bytes(black_box(&contract_bytes)).unwrap());
+}
+
+fn generate_contract(named_keys_len: u8, entry_points_len: u8) -> Contract {
+    let named_keys: NamedKeys = {
+        let mut tmp = BTreeMap::new();
+        (0..named_keys_len).for_each(|i| {
+            let res = tmp.insert(
+                std::str::from_utf8(&[i]).unwrap().to_string(),
+                Key::Account(AccountHash::default()),
+            );
+            assert!(res.is_none());
+        });
+        tmp
+    };
+
+    let entry_points = {
+        let mut tmp = EntryPoints::default();
+        (1..entry_points_len).for_each(|i| {
+            let args = vec![
+                Parameter::new("first", CLType::U32),
+                Parameter::new("Foo", CLType::U32),
+            ];
+            let entry_point = EntryPoint::new(
+                format!("test-{}", i),
+                args,
+                casper_types::CLType::U512,
+                EntryPointAccess::groups(&["Group 2"]),
+                EntryPointType::Contract,
+            );
+            tmp.add_entry_point(entry_point);
+        });
+        tmp
+    };
+
+    casper_types::contracts::Contract::new(
+        ContractPackageHash::default(),
+        ContractWasmHash::default(),
+        named_keys,
+        entry_points,
+        ProtocolVersion::default(),
+    )
+}
+
+fn contract_version_key_fn(i: u8) -> ContractVersionKey {
+    ContractVersionKey::new(i as u32, i as u32)
+}
+
+fn contract_hash_fn(i: u8) -> ContractHash {
+    ContractHash::new([i; KEY_HASH_LENGTH])
+}
+
+fn sample_map<K: Ord, V, FK, FV>(key_fn: FK, value_fn: FV, count: u8) -> BTreeMap<K, V>
+where
+    FK: Fn(u8) -> K,
+    FV: Fn(u8) -> V,
+{
+    (0..count)
+        .map(|i| {
+            let key = key_fn(i);
+            let value = value_fn(i);
+            (key, value)
+        })
+        .collect()
+}
+
+fn sample_set<K: Ord, F>(fun: F, count: u8) -> BTreeSet<K>
+where
+    F: Fn(u8) -> K,
+{
+    (0..count).map(fun).collect()
+}
+
+fn sample_group(i: u8) -> Group {
+    Group::new(format!("group-{}", std::str::from_utf8(&[i]).unwrap()))
```

Comment:
```suggestion
    Group::new(format!("group-{}", i)),
```

Reply:
Done in [`b07c708` (#2389)](https://github.com/casper-network/casper-node/pull/2389/commits/b07c70853e7d9447e20fd5a17ad635a5b8606ad5).